wordpress-mobile/WordPress-Android
# Prepublishing Nudges : Publish button - Button UI Part 1

Fixes #11604

## Solution
This PR is part one for the overall button solution. 

1. The `uploadPost()` logic in the `EditPostActivity` class exists for uploading a post so that was reused with the button action. 

2. `PostSettingsUtils` has logic that creates a label depending on the current `PostStatus` it was reutilized to determine which text to show on the button. - This will be refactored in an upcoming PR based on the additional button logic that will be needed by the `EditorAction`.

## Testing
1. Create a new post. 
2. Select publish.
3. You should see the publish button with text `Publish Now`. 
4. Once you click it and content exists in the post it should publish. 
----------------------------------- 
1. Create a new post. 
2. Select publish.
3. You should see the publish button. 
4. Change the Publish Date from Immediately to one in the future and it should cause the button content to be changed to `Schedule Now`. 

Publish Now | Schedule Now 
--------|-------
 <kbd><img src="https://user-images.githubusercontent.com/1509205/81105048-ff911600-8ed8-11ea-988a-79811270f09d.png" width="320"></kbd>|      <kbd><img src="https://user-images.githubusercontent.com/1509205/81105055-015ad980-8ed9-11ea-9912-8c1a2588bce8.png" width="320"></kbd>
 
## Concerns

1. When the `Publish Date` is changed to a scheduled time and the bottom sheet is dismissed, the primary action is changed to `Schedule` and that doesn't trigger the bottom sheet. I will add that to another PR. https://github.com/wordpress-mobile/WordPress-Android/issues/11895

## Reviewing 
 Only 1 reviewer is needed but anyone can review. 

## Submitter Checklist
